rapid7_insightvm/komand_rapid7_insightvm/util/shared_resources.py

```diff
@@ -0,0 +1,77 @@
+import json
+from komand.exceptions import PluginException
+
+
+class RequestParams(object):
+    """
+    This class transforms a dictionary of parameters into a list of tuples,
+    and allows for their easy manipulation. Accommodates multiple query parameters needed for sorting/filtering requests.
+    """
+
+    def __init__(self, params: [tuple]):
+        self.params = params
+
+    @classmethod
+    def from_dict(cls, params: dict):
+        params_list = list()
+        for key, item in params.items():
+            params_list.append((key, item))
+        return cls(params=params_list)
+
+    @classmethod
+    def from_tuples(cls, params: [tuple]):
+        return cls(params=params)
+
+    # Allows users to get values from this object in the same way you would get values from a dictionary
+    def __getitem__(self, item):
+        for a in self.params:
+            if item in a[0]:
+                return a[1]
+
+    # Allows users to set values from this object in the same way you would set values from a dictionary
+    def __setitem__(self, key, value):
+        for idx, item in enumerate(self.params):
+            if key in item:
+                self.params[idx] = (key, value)
+            else:
+                self.params.append((key, value))
+
+
+def resource_request_status_code_check(response_text: str, status_code: str) -> None:
+    """
+    Checks for non 2xx status codes and raises an exception if found
+    :param response_text: The text of the response
+    :param status_code: response status code
+    :return: None
+    """
+    _ERRORS = {
+        400: "Bad Request",
+        401: "Unauthorized",
+        404: "Not Found",
+        500: "Internal Server Error",
+        503: "Service Unavailable",
+        000: "Unknown Status Code"
+    }
+
+    _ASSISTANCE = {
+        400: "If this issue persists contact support for assistance.",
+        401: "Ensure that the user name and password are correct.",
+        404: "Ensure that the requested resource exists.",
+        500: "If this issue persists contact support for assistance.",
```

Comment:
```suggestion
        500: "If this issue persists contact support for assistance.",
```
Duplicate strings are ripe for a string constant